src/screens/SettingsScreen.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include <freertos/FreeRTOS.h>
+#include <freertos/semphr.h>
+#include <freertos/task.h>
+
+#include <string>
+#include <vector>
+
+#include "Screen.h"
+
+class CrossPointSettings;
+
+// Structure to hold setting information
+struct SettingInfo {
+  const char* name;       // Display name of the setting
+  bool CrossPointSettings::* valuePtr;  // Pointer to member in CrossPointSettings
```

Comment:
Might be good to consider treating this as a `uint8_t` from day 1, `0` and `1` can be used for boolean checking, but it also opens up setting enums in settings too, for example: left, right, justified rendering.